src/pushsource/_impl/backend/koji_source.py

```diff
@@ -621,6 +622,92 @@ def _push_items_from_vmi_build(self, nvr, meta):
 
         return out
 
+    def _push_items_from_rhcos_build(self, nvr, meta):
+
+        LOG.debug("Looking for core os assembler build on %s, %s", nvr, meta)
+
+        build_id = meta["id"]
+        archives = self._get_archives(build_id)
+
+        for archive in archives:
+            if archive.get("filename") == "meta.json":
+                path = self._pathinfo.typedir(meta, archive["btype"])
+                item_src = os.path.join(path, archive["filename"])
+                with open(item_src) as archive_file:
+                    rhcos_meta_data = json.loads(archive_file.read())
+
+        extra = meta.get("extra") or {}
+        image = extra.get("typeinfo").get("image") or {}
+
+        arch = image.get("arch")
+        boot_mode = image.get("boot_mode")
+
+        if boot_mode is not None:
+            boot_mode = BootMode(boot_mode)
+
+        # Try to get the resping version from release, if available
+        respin = try_int(meta["release"])
+        respin = 0 if not isinstance(respin, int) else respin
+
+        vms_with_custom_meta_data = []
+        # add aws ami
+        for ami in rhcos_meta_data["amis"]:
+            vms_with_custom_meta_data.append(
+                {
+                    "target": "aws",
+                    "push_item_class": AmiPushItem,
+                    "custom_meta_data": {
+                        "region": ami.get("name"),
+                        "src": ami.get("hvm"),
+                    },
+                }
+            )
+
+        # add vm for azure
+        vms_with_custom_meta_data.append(
+            {
+                "target": "azure",
+                "push_item_class": VHDPushItem,
+                "custom_meta_data": {
+                    "src": rhcos_meta_data["azure"]["url"],
+                },
+            }
+        )
+
+        out = []
+        for vm_item in vms_with_custom_meta_data:
+            klass = vm_item.get("push_item_class")
+            rel_klass = klass._RELEASE_TYPE
+            release = rel_klass(
+                arch=arch,
+                date=(meta.get("completion_time") or "").split(" ")[0],
+                product=meta.get("name").split("-")[0].upper(),
+                version=meta["version"],
+                respin=respin,
+            )
+
+            out.append(
+                klass(
+                    name=rhcos_meta_data.get("name"),
+                    description=rhcos_meta_data.get("summary"),
+                    **vm_item["custom_meta_data"],
+                    boot_mode=boot_mode,
+                    build=nvr,
+                    build_info=KojiBuildInfo(
+                        id=int(build_id),
+                        name=meta["name"],
+                        version=meta["version"],
+                        release=meta["release"],
+                    ),
+                    sha256sum=rhcos_meta_data.get("images")
```

Comment:
This will crash badly if any of the sums are missing in this "image" structure.  Is it acceptable?

Reply:
If you have easy access to it then I'd suggest including the sha256sum even without a specific use-case in your own tools.  It's one of the attributes on the PushItem base class and Pub understands how to index it, so if you produce this attribute (and do a few other things right) then you can search for push tasks involving that checksum, which can be useful when handling tickets.

Reply:
PTAL